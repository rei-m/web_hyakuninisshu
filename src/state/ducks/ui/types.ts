import { Action } from 'redux';
import { Color, Kimariji } from '@src/types';
import * as constants from './constants';

export type OPEN_KARUTAS_FILTER_TYPE = typeof constants.OPEN_KARUTAS_FILTER_NAME;

export type CLOSE_KARUTAS_FILTER_TYPE = typeof constants.CLOSE_KARUTAS_FILTER_NAME;

export type TOGGLE_KARUTAS_COLOR_TYPE = typeof constants.TOGGLE_KARUTAS_COLOR_NAME;

export type TOGGLE_KARUTAS_KIMARIJI_TYPE = typeof constants.TOGGLE_KARUTAS_KIMARIJI_NAME;

export interface OpenKarutasFilterAction extends Action {
  readonly type: OPEN_KARUTAS_FILTER_TYPE;
}

export interface CloseKarutasFilterAction extends Action {
  readonly type: CLOSE_KARUTAS_FILTER_TYPE;
}

export interface OKarutasFilterAction extends Action {
  readonly type: OPEN_KARUTAS_FILTER_TYPE;
}

export interface ToggleKarutasColorAction extends Action {
  readonly type: TOGGLE_KARUTAS_COLOR_TYPE;
  readonly payload: {
    readonly color: Color;
    readonly checked: boolean;
  };
}

export interface ToggleKarutasKimarijiAction extends Action {
  readonly type: TOGGLE_KARUTAS_KIMARIJI_TYPE;
  readonly payload: {
    readonly kimariji: Kimariji;
    readonly checked: boolean;
  };
}

export type Actions =
  | OpenKarutasFilterAction
  | CloseKarutasFilterAction
  | ToggleKarutasColorAction
  | ToggleKarutasKimarijiAction;

export interface KarutasFilter {
  readonly colors: Array<{ color: Color; checked: boolean }>;
  readonly kimarijis: Array<{ kimariji: Kimariji; checked: boolean }>;
}

export interface UiState {
  readonly karutasFilter: KarutasFilter & { readonly open: boolean };
}
