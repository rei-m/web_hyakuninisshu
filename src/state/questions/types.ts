import { Action } from 'redux';
import { Kimariji, Color, QuestionId, KarutaNo, Karuta } from '@src/domain/models';
import * as constants from './constants';
import { Meta, Payload } from '..';

type Condition<T> = {
  values: Array<T>;
};

export type RangeFromCondition = 1 | 11 | 21 | 31 | 41 | 51 | 61 | 71 | 81 | 91;
export const RangeFromCondition: Condition<RangeFromCondition> = {
  values: [1, 11, 21, 31, 41, 51, 61, 71, 81, 91],
};

export type RangeToCondition = 10 | 20 | 30 | 40 | 50 | 60 | 70 | 80 | 90 | 100;
export const RangeToCondition: Condition<RangeToCondition> = {
  values: [10, 20, 30, 40, 50, 60, 70, 80, 90, 100],
};

export type KimarijiCondition = null | Kimariji;
export const KimarijiCondition: Condition<KimarijiCondition> = {
  values: [null, 1, 2, 3, 4, 5, 6],
};

export type ColorCondition = null | Color;
export const ColorCondition: Condition<ColorCondition> = {
  values: [null, 'blue', 'pink', 'yellow', 'green', 'orange'],
};

export type KarutaStyleCondition = 'kanji' | 'kana';
export const KarutaStyleCondition: Condition<KarutaStyleCondition> = {
  values: ['kanji', 'kana'],
};

export type QuestionAnimCondition = 'none' | 'slow' | 'normal' | 'fast';
export const QuestionAnimCondition: Condition<QuestionAnimCondition> & {
  toDulation: (value: QuestionAnimCondition) => number;
} = {
  values: ['none', 'slow', 'normal', 'fast'],
  toDulation: (value: QuestionAnimCondition) => {
    switch (value) {
      case 'normal':
        return 0.6;
      case 'fast':
        return 0.3;
      case 'slow':
        return 1.0;
      default:
        return 0;
    }
  },
};

export type YomiFuda = Readonly<{
  karutaNo: KarutaNo;
  shoku: string;
  niku: string;
  sanku: string;
}>;

export type ToriFuda = Readonly<{
  karutaNo: KarutaNo;
  shiku: string;
  kekku: string;
}>;

export type StartTrainingAction = Action<typeof constants.START_TRAINING_NAME> &
  Payload<{
    currentQuestionId: QuestionId;
    totalCount: number;
  }> &
  Meta<{
    rangeFrom: RangeFromCondition;
    rangeTo: RangeToCondition;
    kimariji: KimarijiCondition;
    color: ColorCondition;
    kamiNoKuStyle: KarutaStyleCondition;
    shimoNoKuStyle: KarutaStyleCondition;
    questionAnim: QuestionAnimCondition;
  }>;

export type RestartTrainingAction = Action<typeof constants.RESTART_TRAINING_NAME> &
  Payload<{
    currentQuestionId: QuestionId;
    totalCount: number;
  }>;

export type StartExamAction = Action<typeof constants.START_EXAM_NAME> &
  Payload<{
    currentQuestionId: QuestionId;
    totalCount: number;
  }>;

export type StartQuestionAction = Action<typeof constants.START_QUESTION_NAME> &
  Payload<{
    questionId: QuestionId;
    content: {
      yomiFuda: YomiFuda;
      toriFudaList: [ToriFuda, ToriFuda, ToriFuda, ToriFuda];
    };
  }>;

export type ResetQuestionAction = Action<typeof constants.RESET_QUESTION_NAME>;

export type FinishQuestionAction = Action<typeof constants.FINISH_QUESTION_NAME> &
  Payload<{
    correctCount: number;
    averageAnswerSecond: number;
    answerList: Array<{
      questionId: QuestionId;
      isCorrect: boolean;
      correctKaruta: Karuta;
    }>;
  }>;

export type AnswerQuestionAction = Action<typeof constants.ANSWER_QUESTION_NAME> &
  Payload<{
    isCorrect: boolean;
    selectedKarutaNo: KarutaNo;
    correctKaruta: Karuta;
  }>;

export type ConfirmCorrectAction = Action<typeof constants.CONFIRM_CORRECT_NAME>;

export type OpenNextQuestionAction = Action<typeof constants.OPEN_NEXT_QUESTION_NAME> &
  Payload<{
    currentQuestionId: QuestionId;
  }>;

export type Actions =
  | StartTrainingAction
  | RestartTrainingAction
  | StartExamAction
  | StartQuestionAction
  | ResetQuestionAction
  | FinishQuestionAction
  | AnswerQuestionAction
  | ConfirmCorrectAction
  | OpenNextQuestionAction;

export type ActionCreator = {
  /**
   * 指定した条件から問題を作成/保存してStartTrainingActionを返す
   *
   * @param rangeFrom 出題範囲From
   * @param rangeTo 出題範囲To
   * @param kimariji 決まり字条件
   * @param color 色条件
   * @param kamiNoKuStyle 上の句表示形式
   * @param shimoNoKuStyle 下の句表示形式
   * @param questionAnim 問題の表示速度
   *
   * @returns {StartTrainingAction}
   * @throws {Error} 指定した条件に当てはまる歌が無い場合
   */
  startTraining: (
    rangeFrom: RangeFromCondition,
    rangeTo: RangeToCondition,
    kimariji: KimarijiCondition,
    color: ColorCondition,
    kamiNoKuStyle: KarutaStyleCondition,
    shimoNoKuStyle: KarutaStyleCondition,
    questionAnim: QuestionAnimCondition
  ) => StartTrainingAction;

  /**
   * 回答済の問題の中から間違えた問題をもとに問題を作成/保存してRestartTrainingActionを返す
   *
   * @returns {RestartTrainingAction}
   * @throws {Error} 指定した条件に当てはまる歌が無い場合
   */
  restartTraining: () => RestartTrainingAction;

  /**
   * 全ての歌を対象に問題を作成/保存してStartExamActionを返す
   *
   * @returns {StartExamAction}
   * @throws {Error} 指定した条件に当てはまる歌が無い場合
   */
  startExam: () => StartExamAction;

  /**
   * 問題の回答を開始してStartQuestionActionを返す
   *
   * @returns {StartQuestionAction}
   * @throws {Error} 指定した問題がない場合
   */
  startQuestion: (
    currentQuestionId: QuestionId,
    kamiNoKuStyle: KarutaStyleCondition,
    shimoNoKuStyle: KarutaStyleCondition,
    startDate: Date
  ) => StartQuestionAction;

  /**
   * 問題の回答を判定してAnswerQuestionActionを返す
   *
   * @returns {AnswerQuestionAction}
   * @throws {Error} 指定した問題がない場合
   * @throws {Error} 問題の回答が開始されていない場合
   */
  answerQuestion: (currentQuestionId: QuestionId, toriFuda: ToriFuda, answerDate: Date) => AnswerQuestionAction;

  /**
   * ConfirmCorrectActionを返す
   *
   * @returns {ConfirmCorrectAction}
   */
  confirmCorrect: () => ConfirmCorrectAction;

  /**
   * 次の問題を取得してOpenNextQuestionActionを返す
   *
   * @returns {OpenNextQuestionAction}
   * @throws {Error} 次の問題がない場合
   */
  openNextQuestionAction: (currentQuestionId: QuestionId) => OpenNextQuestionAction;

  /**
   * 問題を初期状態に戻してResetQuestionActionを返す
   *
   * @returns {ResetQuestionAction}
   */
  resetQuestion: () => ResetQuestionAction;

  /**
   * 問題を終了して結果を見る
   *
   * @returns {FinishQuestionAction}
   */
  finishQuestion: () => FinishQuestionAction;
};

export type State = Readonly<{
  state: 'waiting' | 'ready' | 'playing' | 'confirm' | 'finished';
  totalCount: number;
  currentPosition: number;
  currentQuestion?: {
    questionId: QuestionId;
    content?: {
      yomiFuda: YomiFuda;
      toriFudaList: [ToriFuda, ToriFuda, ToriFuda, ToriFuda];
    };
    answer?: {
      isCorrect: boolean;
      selectedKarutaNo: KarutaNo;
      correctKaruta: Karuta;
    };
  };
  result?: {
    correctCount: number;
    averageAnswerSecond: number;
    answerList: Array<{
      questionId: QuestionId;
      isCorrect: boolean;
      correctKaruta: Karuta;
    }>;
  };
  trainingCondition: {
    rangeFrom: RangeFromCondition;
    rangeTo: RangeToCondition;
    kimariji: KimarijiCondition;
    color: ColorCondition;
    kamiNoKuStyle: KarutaStyleCondition;
    shimoNoKuStyle: KarutaStyleCondition;
    questionAnim: QuestionAnimCondition;
  };
}>;
