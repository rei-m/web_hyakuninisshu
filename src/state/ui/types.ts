import { Action } from 'redux';
import * as constants from './constants';
import { Color, Kimariji } from '@src/domain/models';
import { Payload } from '..';

export type KarutasFilter = Readonly<{
  colors: Array<{ color: Color; checked: boolean }>;
  kimarijis: Array<{ kimariji: Kimariji; checked: boolean }>;
}>;

export type OpenKarutasFilterAction = Action<typeof constants.OPEN_KARUTAS_FILTER_NAME>;

export type CloseKarutasFilterAction = Action<typeof constants.CLOSE_KARUTAS_FILTER_NAME>;

export type ToggleKarutasColorAction = Action<typeof constants.TOGGLE_KARUTAS_COLOR_NAME> &
  Payload<{
    color: Color;
    checked: boolean;
  }>;

export type ToggleKarutasKimarijiAction = Action<typeof constants.TOGGLE_KARUTAS_KIMARIJI_NAME> &
  Payload<{
    kimariji: Kimariji;
    checked: boolean;
  }>;

export type Actions =
  | OpenKarutasFilterAction
  | CloseKarutasFilterAction
  | ToggleKarutasColorAction
  | ToggleKarutasKimarijiAction;

export type ActionCreator = {
  /**
   * OpenKarutasFilterActionを返す
   *
   * @returns {OpenKarutasFilterAction}
   */
  openKarutasFilter: () => OpenKarutasFilterAction;

  /**
   * CloseKarutasFilterActionを返す
   *
   * @returns {CloseKarutasFilterAction}
   */
  closeKarutasFilter: () => CloseKarutasFilterAction;

  /**
   * ToggleKarutasColorActionを返す
   *
   * @param color 歌の色
   * @param checked 表示する/しない
   *
   * @returns {ToggleKarutasColorAction}
   */
  toggleKarutasColor: (color: Color, checked: boolean) => ToggleKarutasColorAction;

  /**
   * ToggleKarutasKimarijiActionを返す
   *
   * @param kimariji 歌の決まり字
   * @param checked 表示する/しない
   *
   * @returns {ToggleKarutasKimarijiAction}
   */
  toggleKarutasKimariji: (kimariji: Kimariji, checked: boolean) => ToggleKarutasKimarijiAction;
};

export type State = Readonly<{
  karutasFilter: KarutasFilter & { open: boolean };
}>;
