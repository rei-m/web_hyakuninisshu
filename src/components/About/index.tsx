import * as React from 'react';

const About = () => {
  return <div>サイトについて</div>;
};

export default About;
