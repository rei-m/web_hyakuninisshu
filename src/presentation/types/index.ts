import { RouteComponentProps } from '@reach/router';

export type SiteMetaData = {
  title: string;
  description: string;
  author: string;
};

export type GeneratedPageComponentProps<T = {}> = { pageContext: T } & RouteComponentProps<T>;

export type AppErrorType = 'Network' | 'NotFound' | 'Unknown';

export class AppError implements Error {
  private _message: string;
  private _type: AppErrorType;

  constructor(message: string, type: AppErrorType) {
    this._message = message;
    this._type = type;
  }

  public get name() {
    return 'AppError';
  }

  public get message() {
    return this._message;
  }

  public get type() {
    return this._type;
  }

  public toString() {
    return `${this.name}: ${this.message}`;
  }
}
