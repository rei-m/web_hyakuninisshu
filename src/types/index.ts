export interface Karuta {
  id: number;
  imageNo: string;
}

export interface Question {
  correctKaruta: Karuta;
  choices: Karuta[];
}

export interface OptionItem<T> {
  readonly name: string;
  readonly value: T;
}
