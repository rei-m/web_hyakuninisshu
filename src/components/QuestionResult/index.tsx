import * as React from 'react';
import styled from 'styled-components';
import { Answer } from '../../types';
import * as correctImage from './check_correct.png';
import * as incorrectImage from './check_incorrect.png';

export interface QuestionResultProps {
  answer: Answer;
  onClick: () => void;
}

const Frame = styled.div`
  background-color: transparent;
  position: absolute;
  left: 0;
  top: 0;
  width: 100vw;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const Image = styled.img`
  width: 300px;
  height: 300px;
`;

const QuestionResult = ({ answer, onClick }: QuestionResultProps) => {
  const onClickFrame = (_: React.SyntheticEvent<HTMLDivElement>) => {
    onClick();
  };
  return (
    <Frame onClick={onClickFrame}>
      {answer.correct ? (
        <Image src={correctImage} />
      ) : (
        <Image src={incorrectImage} />
      )}
    </Frame>
  );
};

export default QuestionResult;
