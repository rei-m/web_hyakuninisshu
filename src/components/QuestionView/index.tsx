import * as React from 'react';
import { graphql, StaticQuery } from 'gatsby';
import styled from '@src/styles/styled-components';
import YomiFudaView from '@src/components/YomiFudaView';
import ToriFudaView from '@src/components/ToriFudaView';
import QuestionResult from '@src/components/QuestionResult';
import { Answer, Question, ToriFuda } from '@src/types';

export interface Props {
  question: Question;
  answer?: Answer;
  totalCount: number;
  currentPosition: number;
  onClickToriFuda: (toriFuda: ToriFuda) => void;
  onClickResult: () => void;
}

interface QueryData {
  questionBGImage: {
    publicURL: string;
  };
}

const Container = styled.div<{ bgImageUrl: string }>`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-image: url("${({ bgImageUrl }) => bgImageUrl}");
  min-height: calc(100vh - ${({ theme }) => theme.headerHeight});
  @media screen and (min-width: ${({ theme }) => theme.minWidthWide}) {
    min-height: calc(100vh - ${({ theme }) => theme.headerHeightWide});
  }
`;

const YomiFudaBox = styled.div`
  position: relative;
  margin-bottom: ${({ theme }) => theme.spacing3x};
  width: 312px;
`;

const ToriFudaBox = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: row-reverse;
`;

const Position = styled.div`
  position: absolute;
  right: 0;
  top: -32px;
  color: #fff;
  border-bottom: 1px dotted #fff;
`;

const QuestionView: React.FC<Props> = ({
  answer,
  currentPosition,
  question,
  totalCount,
  onClickToriFuda,
  onClickResult,
}) => (
  <StaticQuery
    query={query}
    render={({ questionBGImage }: QueryData) => (
      <Container bgImageUrl={questionBGImage.publicURL}>
        <YomiFudaBox>
          <Position>
            {currentPosition} / {totalCount}
          </Position>
          <YomiFudaView yomiFuda={question.yomiFuda} style={{ margin: 'auto' }} />
        </YomiFudaBox>
        <ToriFudaBox>
          {question.toriFudas.map((toriFuda, i) => (
            <ToriFudaView
              toriFuda={toriFuda}
              style={{
                margin: '8px',
                opacity: !answer || answer.karutaNo === toriFuda.karutaNo ? 1 : 0.8,
              }}
              key={i}
              onClick={onClickToriFuda}
              data-test={`torifuda-${i}`}
            />
          ))}
        </ToriFudaBox>
        {answer && <QuestionResult answer={answer} onClick={onClickResult} data-test="result" />}
      </Container>
    )}
  />
);

export default QuestionView;

const query = graphql`
  query {
    questionBGImage: file(relativePath: { eq: "tatami_part.png" }) {
      publicURL
    }
  }
`;
