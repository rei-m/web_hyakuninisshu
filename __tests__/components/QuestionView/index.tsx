import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import { shallow } from 'enzyme';
import { create } from '@test/factories';
import { sel, setUpQuery } from '@test/helpers';
import QuestionView, { Props, QueryData as QuestionViewQueryData } from '@src/components/QuestionView';
import { QueryData as QuestionResultQueryData } from '@src/components/QuestionResult';
import { Answer, Question } from '@src/types';

describe('<QuestionView />', () => {
  let baseProps: Props;

  beforeEach(() => {
    setUpQuery<QuestionViewQueryData & QuestionResultQueryData>({
      correctImage: {
        childImageSharp: {
          fluid: {
            aspectRatio: 1,
            sizes: `100 200 300`,
            src: `correct-base64-encoded-image`,
            srcSet: `asdfasdf`,
          },
        },
      },
      incorrectImage: {
        childImageSharp: {
          fluid: {
            aspectRatio: 1,
            sizes: `100 200 300`,
            src: `incorrect-base64-encoded-image`,
            srcSet: `asdfasdf`,
          },
        },
      },
      questionBGImage: {
        publicURL: '/tatami.png',
      },
    });

    baseProps = {
      currentPosition: 1,
      onClickResult: jest.fn(),
      onClickToriFuda: jest.fn(),
      question: create<Question>('question'),
      totalCount: 10,
      dulation: 0.6,
    };
  });

  it('should render component when not answered', () => {
    const renderer = ReactTestRenderer.create(<QuestionView {...baseProps} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should render component when answered', () => {
    const props = {
      ...baseProps,
      answer: create<Answer>('answer', { questionId: baseProps.question.id }),
    };
    const renderer = ReactTestRenderer.create(<QuestionView {...props} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should fire onClickToriFuda when torifuda clicked', () => {
    const wrapper = shallow(<QuestionView {...baseProps} />).dive();
    wrapper
      .find(sel('torifuda-0'))
      .dive()
      .simulate('click');
    expect(baseProps.onClickToriFuda).toHaveBeenCalledWith(baseProps.question.toriFudas[0]);
  });

  it('should fire onClickResult when result clicked', () => {
    const props = {
      ...baseProps,
      answer: create<Answer>('answer', { questionId: baseProps.question.id }),
    };
    const wrapper = shallow(<QuestionView {...props} />).dive();
    wrapper.find(sel('result')).simulate('click');
    expect(baseProps.onClickResult).toHaveBeenCalled();
  });
});
