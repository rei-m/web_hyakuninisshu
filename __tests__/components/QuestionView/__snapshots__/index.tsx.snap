// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionView /> should render component when answered 1`] = `
<div
  className="sc-VigVT ljONEf"
>
  <div
    className="sc-jTzLTM jnXhag"
  >
    <div
      className="sc-jzJRlG hbbPSk"
    >
      1
       / 
      10
    </div>
    <div
      className="sc-bwzfXH bvmjNt"
      style={
        Object {
          "margin": "auto",
        }
      }
    >
      <div
        className="sc-htpNat eaelvy"
      >
        <div
          className="sc-bxivhb cIIvqJ"
        >
          初句
        </div>
        <div
          className="sc-bxivhb sc-ifAKCX ejiZuo"
        >
          二句
        </div>
        <div
          className="sc-bxivhb sc-EHOje fUTEIU"
        >
          三句
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-fjdhpX hZnkOe"
  >
    <div
      className="sc-bZQynM lmXxAo"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <div
        className="sc-gzVnrw bvFKOA"
        color="#fff"
        onClick={[Function]}
        style={
          Object {
            "margin": "8px",
            "opacity": 0.8,
          }
        }
      >
        <div
          className="sc-htoDjs bKhwRD"
        >
          <div
            className="sc-dnqmqq gCITGr"
          >
            よんく
          </div>
          <div
            className="sc-dnqmqq sc-iwsKbI cSrnkn"
          >
            ごく
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bZQynM lmXxAo"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <div
        className="sc-gzVnrw bvFKOA"
        color="#fff"
        onClick={[Function]}
        style={
          Object {
            "margin": "8px",
            "opacity": 0.8,
          }
        }
      >
        <div
          className="sc-htoDjs bKhwRD"
        >
          <div
            className="sc-dnqmqq gCITGr"
          >
            よんく
          </div>
          <div
            className="sc-dnqmqq sc-iwsKbI cSrnkn"
          >
            ごく
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bZQynM lmXxAo"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <div
        className="sc-gzVnrw bvFKOA"
        color="#fff"
        onClick={[Function]}
        style={
          Object {
            "margin": "8px",
            "opacity": 1,
          }
        }
      >
        <div
          className="sc-htoDjs bKhwRD"
        >
          <div
            className="sc-dnqmqq gCITGr"
          >
            よんく
          </div>
          <div
            className="sc-dnqmqq sc-iwsKbI cSrnkn"
          >
            ごく
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bZQynM lmXxAo"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <div
        className="sc-gzVnrw bvFKOA"
        color="#fff"
        onClick={[Function]}
        style={
          Object {
            "margin": "8px",
            "opacity": 0.8,
          }
        }
      >
        <div
          className="sc-htoDjs bKhwRD"
        >
          <div
            className="sc-dnqmqq gCITGr"
          >
            よんく
          </div>
          <div
            className="sc-dnqmqq sc-iwsKbI cSrnkn"
          >
            ごく
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-gZMcBi bKdPNs"
    onClick={[MockFunction]}
  >
    <div
      className="sc-gqjmRU kzcJUc gatsby-image-wrapper"
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": "100%",
            "width": "100%",
          }
        }
      />
      <picture>
        <source
          sizes="100 200 300"
          srcSet="asdfasdf"
        />
        <img
          alt=""
          onLoad={[Function]}
          src="correct-base64-encoded-image"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 0.5s",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcSet=\\"asdfasdf\\" sizes=\\"100 200 300\\" /><img src=\\"correct-base64-encoded-image\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<QuestionView /> should render component when not answered 1`] = `
<div
  className="sc-VigVT ljONEf"
>
  <div
    className="sc-jTzLTM jnXhag"
  >
    <div
      className="sc-jzJRlG hbbPSk"
    >
      1
       / 
      10
    </div>
    <div
      className="sc-bwzfXH bvmjNt"
      style={
        Object {
          "margin": "auto",
        }
      }
    >
      <div
        className="sc-htpNat eaelvy"
      >
        <div
          className="sc-bxivhb cIIvqJ"
        >
          <span
            className="sc-bdVaJa siDlf"
            onAnimationEnd={[Function]}
          >
            初
          </span>
          <span>
            　
          </span>
        </div>
        <div
          className="sc-bxivhb sc-ifAKCX ejiZuo"
        >
          <span>
            　
          </span>
          <span>
            　
          </span>
        </div>
        <div
          className="sc-bxivhb sc-EHOje fUTEIU"
        >
          <span>
            　
          </span>
          <span>
            　
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-fjdhpX hZnkOe"
  >
    <div
      className="sc-bZQynM lmXxAo"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <div
        className="sc-gzVnrw bvFKOA"
        color="#fff"
        onClick={[Function]}
        style={
          Object {
            "margin": "8px",
            "opacity": 1,
          }
        }
      >
        <div
          className="sc-htoDjs bKhwRD"
        >
          <div
            className="sc-dnqmqq gCITGr"
          >
            よんく
          </div>
          <div
            className="sc-dnqmqq sc-iwsKbI cSrnkn"
          >
            ごく
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bZQynM lmXxAo"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <div
        className="sc-gzVnrw bvFKOA"
        color="#fff"
        onClick={[Function]}
        style={
          Object {
            "margin": "8px",
            "opacity": 1,
          }
        }
      >
        <div
          className="sc-htoDjs bKhwRD"
        >
          <div
            className="sc-dnqmqq gCITGr"
          >
            よんく
          </div>
          <div
            className="sc-dnqmqq sc-iwsKbI cSrnkn"
          >
            ごく
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bZQynM lmXxAo"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <div
        className="sc-gzVnrw bvFKOA"
        color="#fff"
        onClick={[Function]}
        style={
          Object {
            "margin": "8px",
            "opacity": 1,
          }
        }
      >
        <div
          className="sc-htoDjs bKhwRD"
        >
          <div
            className="sc-dnqmqq gCITGr"
          >
            よんく
          </div>
          <div
            className="sc-dnqmqq sc-iwsKbI cSrnkn"
          >
            ごく
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bZQynM lmXxAo"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <div
        className="sc-gzVnrw bvFKOA"
        color="#fff"
        onClick={[Function]}
        style={
          Object {
            "margin": "8px",
            "opacity": 1,
          }
        }
      >
        <div
          className="sc-htoDjs bKhwRD"
        >
          <div
            className="sc-dnqmqq gCITGr"
          >
            よんく
          </div>
          <div
            className="sc-dnqmqq sc-iwsKbI cSrnkn"
          >
            ごく
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
